main.cpp

```diff
@@ -265,6 +296,21 @@ struct IntType
 
     operator int() const { return *a; }
 
+    IntType& apply( std::function<IntType&( int& )> func )
+    {
+        if ( func != nullptr )
+            func( *a );
+        return *this;
```

Comment:
`func()` returns `IntType&`, so you should return that object.   If you mentally step through the execution, you'll see that the returned object is `*this`.
- [ ] revise.